cfme/tests/distributed/test_appliance_replication.py

```diff
@@ -202,3 +210,60 @@ def test_distributed_vm_power_control(provider, create_vm,
         soft_assert(
             not create_vm.mgmt.is_running,
             "vm running")
+
+
+@pytest.mark.rhel_testing
+@pytest.mark.tier(2)
+@pytest.mark.meta(automates=[1678142])
+@pytest.mark.ignore_stream('upstream', '5.10')
+def test_replication_connect_to_vm_in_region(provider,
+        temp_appliance_preconfig_funcscope_rhevm, temp_appliance_unconfig_funcscope_rhevm):
+    """Test that the user can view the VM in the global appliance UI, click on the
+    "Connect to VM in its Region" button, and be redirected to the VM in the remote appliance UI.
+
+    Metadata:
+        test_flag: replication
+
+    Polarion:
+        assignee: tpapaioa
+        initialEstimate: 1/4h
+        casecomponent: Appliance
+        startsin: 5.11
+    """
+    remote_app = temp_appliance_preconfig_funcscope_rhevm
+    global_app = temp_appliance_unconfig_funcscope_rhevm
+
+    configure_replication_appliances(remote_app, global_app)
+
+    vm_name = provider.data['cap_and_util']['chargeback_vm']
+
+    remote_app.browser_steal = True
+    with remote_app:
+        provider.create()
+        remote_app.collections.infra_providers.wait_for_a_provider()
+
+    global_app.browser_steal = True
+    with global_app:
+        global_collection = global_app.provider_based_collection(provider)
+        vm = global_collection.instantiate(vm_name, provider)
+        view = navigate_to(vm, 'Details')
+        view.entities.summary('Multi Region').click_at('Remote Region')
+        sleep(5)
```

Comment:
Is there any `wait_for` that can be used here? Or since you are going to a new tab is that not possible?

Reply:
There needs to be a wait between switch_to_window() and view.is_displayed, otherwise the browser hangs. I've cleared up the code here a bit, copying some over from cfme/tests/webui/test_general_ui.py. There's a TODO comment about handling WebDriverException in ensure_page_safe(). I haven't looked into what's required for that yet, but it looks like the call to sleep() is the best way for now.